java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/ui/NotifyDescriptorAdapter.java

```diff
@@ -0,0 +1,322 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.accessibility.Accessible;
+import javax.accessibility.AccessibleContext;
+import javax.accessibility.AccessibleIcon;
+import javax.accessibility.AccessibleText;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import org.eclipse.lsp4j.MessageActionItem;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.ShowMessageRequestParams;
+import org.openide.NotifyDescriptor;
+import org.openide.util.NbBundle;
+
+/**
+ * Adapts a {@link NotifyDescriptor} to a {@link ShowMessageRequestParams} call.
+ * @author sdedic
+ */
+class NotifyDescriptorAdapter {
+    private static final Logger LOG = Logger.getLogger(NotifyDescriptorAdapter.class.getName());
+    
+    private final UIContext client;
+    private final NotifyDescriptor  descriptor;
+    private final Map<MessageActionItem, Object> item2Option = new LinkedHashMap<>();
+    private final Map<String, Object> text2Option = new HashMap<>();
+    private final Map<Object, List<ActionListener>> optionListeners = new HashMap<>();
+    private final Map<Object, JButton> option2Button = new HashMap<>();
+    
+    private static final Set<String> warnedClasses = new HashSet<>();
+    
+    private ShowMessageRequestParams request;
+
+    private static final Object[] YES_NO_CANCEL = new Object[] {
+        NotifyDescriptor.YES_OPTION,
+        NotifyDescriptor.NO_OPTION,
+        NotifyDescriptor.CANCEL_OPTION
+    };
+    
+    private static final Object[] YES_NO = new Object[] {
+        NotifyDescriptor.YES_OPTION,
+        NotifyDescriptor.NO_OPTION,
+    };
+    
+    private static final Object[] OK_CANCEL = new Object[] {
+        NotifyDescriptor.OK_OPTION,
+        NotifyDescriptor.CANCEL_OPTION
+    };
+    
+    private static final Object[] JUST_OK = new Object[] {
+        NotifyDescriptor.OK_OPTION
+    };
+    
+    public NotifyDescriptorAdapter(NotifyDescriptor descriptor, UIContext client) {
+        this.descriptor = descriptor;
+        this.client = client;
+    }
+    
+    private MessageType translateMessageType() {
+        switch(descriptor.getMessageType()) {
+            case NotifyDescriptor.ERROR_MESSAGE:
+                return MessageType.Error;
+            case NotifyDescriptor.WARNING_MESSAGE:
+                return MessageType.Warning;
+            
+            case NotifyDescriptor.QUESTION_MESSAGE:
+            case NotifyDescriptor.PLAIN_MESSAGE:
+            case NotifyDescriptor.INFORMATION_MESSAGE:
+                return MessageType.Info;
+            default:
+                return MessageType.Log;
+        }
+    }
+    
+    /**
+     * Strip HTML from the message; VSCode standard showMessage does not support HTML.
+     * @param original
+     * @return 
+     */
+    private String translateText(String original) {
```

Comment:
I believe the translation of the text away from HTML shall only happen if it starts with `<html>`.